opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -887,6 +887,7 @@ namespace Opm
         auto obtain = [this](const Eval& value)
                       {
                           if constexpr (std::is_same_v<Value, Scalar>) {
+                              (void)this; // suppress clang warning
```

Comment:
Would you mind writing `static_cast<>()` please?  That stands out more clearly when reading the code.

Reply:
sure.